### PR TITLE
Add Meld Build Plugin

### PR DESCRIPTION
**Pull request by meld-cp**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/meld-cp/obsidian-build

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
[interface MeldBuildPluginSettings {](https://github.com/meld-cp/obsidian-build/blob/24a3de716a65839870747ba5226d5c5a4945f881/src/main.ts#L7-L13)
Since you are not using settings you can remove this and any references to it.

[while(!this.completed){](https://github.com/meld-cp/obsidian-build/blob/24a3de716a65839870747ba5226d5c5a4945f881/src/modal-ask.ts#L33-L37)
I would recommend using a callback approach or similar here, instead of constantly waiting.

[app](https://github.com/meld-cp/obsidian-build/blob/24a3de716a65839870747ba5226d5c5a4945f881/src/rci-io.ts#L101)
It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future. Instead, you should use the reference to App that you have from your plugin instance.

[public static delay(ms: number) : Promise<()=>void> {](https://github.com/meld-cp/obsidian-build/blob/24a3de716a65839870747ba5226d5c5a4945f881/src/utils.ts#L23-L25)
The Obsidian already provide a `sleep` method for this, which does basically the same.

**Comment by meld-cp** [comment]
Thanks for the review, those changes have been made.

**Comment by liamcain** [comment]
- [innerHTML](https://github.com/meld-cp/obsidian-build/blob/71c45bdf0790959a7a15833742dd60c86ba37a20/src/modal-message.ts#L35) [innerHTML is a potential security risk](https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md#avoid-innerhtml-outerhtml-and-insertadjacenthtml). It's hard to trace through the code to see what's creating these messages. If it's something you control, you should just create the html yourself. If it's generated by the user, then I suggest limiting it to strings and using setText()
- [read](https://github.com/meld-cp/obsidian-build/blob/71c45bdf0790959a7a15833742dd60c86ba37a20/src/rci-io.ts#L97) if you aren't writing the file contents back to the disk, you should use [cachedRead](https://github.com/obsidianmd/obsidian-api/blob/9bb30633586c6649474b2affdf5b87efdd86498f/obsidian.d.ts#L3373) instead. It's faster since you're just reading the file contents from memory.

**Comment by meld-cp** [comment]
Thank you!

- [innerHTML](https://github.com/meld-cp/obsidian-build/commit/214df0d6fb92c09efa810c9b82e2630d86974155) has been replaced
- [read replaced with cachedRead](https://github.com/meld-cp/obsidian-build/commit/26181f02dc48685bbff797cb82cc7c33e6bf2946)

File: community-plugins.json [hunk 1]
```diff
@@ -6386,5 +6386,13 @@
...
         "author": "froehlichA",
         "repo": "froehlichA/obsidian-avatar"
+    },
+    {
+        "id": "meld-build",
+        "name": "Meld Build",
+        "description": "Write and execute (sandboxed) JavaScript to render templates, query DataView and create dynamic notes.",
+        "author": "meld-cp",
+        "repo": "meld-cp/obsidian-build",
+        "branch": "master"
```
Comment (liamcain) [inline community-plugins.json:6396]:
```suggestion
        "repo": "meld-cp/obsidian-build"
```

**Comment by meld-cp** [comment]
Thank you guys 👍

**Pull request merged**